dotnet-trace won't run in Kudu console

Repro:
1. Create an Azure App Service for Windows
1. (Optional) Deploy an ASP.NET Core 3.0 app to the App Service
1. Open the Kudu site for your App Service in the Browser (Find "Advanced Tools" from the App in the Azure Portal)
1. Go to the Process Explorer and find the PID of your app (it's probably the w3wp.exe process)
1. Click on Debug Console and select Cmd
1. At the cmd prompt, type:
1. `dotnet tool install --global dotnet-trace`
1. There's a warning that the tools directory is not on the PATH. So run
    `set PATH=%PATH%;%USERPROFILE%\.dotnet\tools`
1. `dotnet-trace collect --profile gc-verbose -p 4124 -o gc-verbose.netperf` (substituting the PID from the earlier step for 4124)

Result:
```
[ERROR] System.IO.IOException: The handle is invalid
   at System.ConsolePal.GetBufferInfo(Boolean throwOnNoConsole, Boolean& succeeded)
   at System.ConsolePal.Clear()
   at Microsoft.Diagnostics.Tools.Trace.CollectCommandHandler.Collect(CancellationTo
ken ct, IConsole console, Int32 processId, FileInfo output, UInt32 buffersize, String providers, String profile, TraceFileFormat format, TimeSpan duration) in /_/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs:line 41
```

Expect:
Should be able to do a collection on Antares from the Kudu window.

--- a/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs
+++ b/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs
@@ -38,7 +38,16 @@ namespace Microsoft.Diagnostics.Tools.Trace
             {
                 Debug.Assert(output != null);
                 Debug.Assert(profile != null);
-                Console.Clear();
+
+                try
+                {
+                    Console.Clear();
+                }
+                catch (IOException)
+                {
+                    // Console not available.
+                }
+
                 if (processId < 0)
                 {
                     Console.Error.WriteLine("Process ID should not be negative.");

--- a/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs
+++ b/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs
@@ -39,13 +39,17 @@ namespace Microsoft.Diagnostics.Tools.Trace
                 Debug.Assert(output != null);
                 Debug.Assert(profile != null);
 
+                bool hasConsole = true;
+
                 try
                 {
                     Console.Clear();
                 }
                 catch (IOException)
                 {
-                    // Console not available.
+                    // The console is not available, probably because it has been redirected.
+                    // See https://docs.microsoft.com/dotnet/api/system.console.clear#remarks
+                    hasConsole = false;
                 }
 
                 if (processId < 0)
@@ -150,8 +154,13 @@ namespace Microsoft.Diagnostics.Tools.Trace
                                     if (nBytesRead <= 0)
                                         break;
                                     fs.Write(buffer, 0, nBytesRead);
-                                    lineToClear = Console.CursorTop-1;
-                                    ResetCurrentConsoleLine(vTermMode.IsEnabled);
+
+                                    if (hasConsole)
+                                    {
+                                        lineToClear = Console.CursorTop - 1;
+                                        ResetCurrentConsoleLine(vTermMode.IsEnabled);
+                                    }
+
                                     Console.Out.WriteLine($"[{stopwatch.Elapsed.ToString(@"dd\:hh\:mm\:ss")}]\tRecording trace {GetSize(fs.Length)}");
                                     Console.Out.WriteLine("Press <Enter> or <Ctrl+C> to exit...");
                                     Debug.WriteLine($"PACKET: {Convert.ToBase64String(buffer, 0, nBytesRead)} (bytes {nBytesRead})");

--- a/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs
+++ b/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs
@@ -39,18 +39,10 @@ namespace Microsoft.Diagnostics.Tools.Trace
                 Debug.Assert(output != null);
                 Debug.Assert(profile != null);
 
-                bool hasConsole = true;
+                bool hasConsole = console.GetTerminal() != null;
 
-                try
-                {
+                if (hasConsole)
                     Console.Clear();
-                }
-                catch (IOException)
-                {
-                    // The console is not available, probably because it has been redirected.
-                    // See https://docs.microsoft.com/dotnet/api/system.console.clear#remarks
-                    hasConsole = false;
-                }
 
                 if (processId < 0)
                 {

